src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/Kavita.kt

```diff
@@ -513,11 +513,14 @@ class Kavita(private val suffix: String = "") : ConfigurableSource, UnmeteredSou
      * Fetches the "url" of each page from the chapter
      * **/
     override fun pageListRequest(chapter: SChapter): Request {
-        return GET("$apiUrl/${chapter.url}", headersBuilder().build())
+        // remove potential _<part> chapter salt
+        val chapterId = chapter.url.substringBefore("_")
+        return GET("$apiUrl/$chapterId", headersBuilder().build())
```

Comment:
@majora2007 we sure there won't ever be other underscores apart from the salt one we add, right? Otherwise seems fine.

Reply:
What does the url map to? A special can technically have _ in the name.

Reply:
url is defined as url = chapter.id.toString().

Reply:
So maybe I missed it, how is the underscore being sent?

Reply:
It’s not, I’m adding it locally.
<chapterId>_<numberOfFiles>
As ThePromidius points out, this approach would fail if the id somehow already contains an underscore.